File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 6/27/2025, 2:30:59 AM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curSum = nums[0];
        int maxSum = nums[0];
        for(int i = 1; i < nums.size(); i++){
            curSum = max(nums[i], curSum + nums[i]);
            maxSum = max(curSum, maxSum);
        }  
        return maxSum;
    }
};

File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 6/27/2025, 2:31:03 AM
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums.length == 1) {
            return nums[0];  // Handle single-element case
        }

        int currentSum = nums[0];  // Start with the first element
        int maxSum = nums[0];      // Start with the first element

        for (int index = 1; index < nums.length; index++) {
            currentSum = Math.max(nums[index], currentSum + nums[index]);
            maxSum = Math.max(currentSum, maxSum);
        }

        return maxSum;
    }
}